keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,302 @@
+from tensorflow import keras
+
+from keras_nlp.layers import TokenAndPositionEmbedding
+from keras_nlp.layers import TransformerEncoder
+
+
+def _roberta_kernel_initializer(stddev=0.02):
+    return keras.initializers.TruncatedNormal(stddev=stddev)
+
+
+class Roberta(keras.Model):
+    """Robustly optimized Bert pretraining approach.
+
+    This network implements a bi-directional Transformer-based encoder as
+    described in ["RoBERTa: A Robustly Optimized BERT Pretraining Approach"](https://arxiv.org/abs/1907.11692).
+    It includes the embedding lookups and transformer layers, but not include
+    the masked language model network.
+
+    This class gives a fully configurable Roberta model with any number of
+    layers, heads, and embedding dimensions. For specific roberta architectures
+    defined in the paper, see for example `keras_nlp.models.RobertaBase`.
+
+    Args:
+        vocabulary_size: Int. The size of the token vocabulary.
+        num_layers: Int. The number of transformer layers.
+        num_heads: Int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: Int. The size of the transformer encoding layer.
+        intermediate_dim: Int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        dropout: Float. Dropout probability for the Transformer encoder.
+        max_sequence_length: Int. The maximum sequence length that this encoder
+            can consume. If None, `max_sequence_length` uses the value from
+            sequence length. This determines the variable shape for positional
+            embeddings.
+        bos_token_index: Int. Index of <s> token in the vocabulary. Equivalent
+            to [CLS] in BERT.
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```python
+    # Randomly initialized Roberta encoder
+    encoder = keras_nlp.models.Roberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call encoder on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=30522),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    output = encoder(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.1,
+        max_sequence_length=512,
+        bos_token_index=0,
+        name=None,
+        trainable=True,
+    ):
+
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="input_ids"
```

Comment:
we are using two different style of dtype (tf.x or "x"), in this file. let's chose one and be consistent.